tools/github-event-processor/Azure.Sdk.Tools.GitHubEventProcessor/EventProcessing/IssueProcessing.cs

```diff
@@ -81,25 +89,140 @@ public static async Task InitialIssueTriage(GitHubEventClient gitHubEventClient,
                             foreach (string label in labelSuggestions)
                             {
                                 gitHubEventClient.AddLabel(label);
                             }
-                            gitHubEventClient.AddLabel(LabelConstants.NeedsTeamTriage);
+
+                            var azureSdkOwners = CodeOwnerUtils.GetAzureSdkOwnersForServiceLabels(labelSuggestions);
+                            bool hasValidAssignee = false;
+                            if (azureSdkOwners.Count > 0)
+                            {
+                                // If there's only a single owner, 
+                                if (azureSdkOwners.Count == 1)
+                                {
+                                    if (await gitHubEventClient.OwnerCanBeAssignedToIssuesInRepo(
+                                                                    issueEventPayload.Repository.Owner.Login,
+                                                                    issueEventPayload.Repository.Name,
+                                                                    azureSdkOwners[0]))
+                                    {
+                                        hasValidAssignee = true;
+                                        gitHubEventClient.AssignOwnerToIssue(
+                                                            issueEventPayload.Repository.Owner.Login,
+                                                            issueEventPayload.Repository.Name,
+                                                            azureSdkOwners[0]);
+                                    }
+                                    // Output something into the logs pointing out that AzureSdkOwners has a user that cannot
+                                    // be assigned to an issue
+                                    else
+                                    {
+                                        Console.WriteLine($"{azureSdkOwners[0]} is the only owner in the AzureSdkOwners for service label(s), {string.Join(",", labelSuggestions)}, but cannot be assigned as an issue owner in this repository.");
+                                    }
+                                }
+                                // else there are multiple owners and a random one needs to be assigned
+                                else
+                                {
+                                    // Create a list of AzureSdkOwners that has been randomized. The reason
+                                    // the entire list is being randomed is because each person has to be
+                                    // checked to see if they can be assigned to an issue in the repository.
+                                    // and having the entire list being random simplifies processing if a given
+                                    // owner cannot be assigned.
+                                    var rnd = new Random();
```

Comment:
@jsquire seeding this doesn't actually matter. The way it's written, right now, it'll completely randomize the list each time it runs.  There is no place to save a seed nor was there any requirement to try and round robin assignment. That can't actually happen with Actions since there's no place to save it and honestly, that would cause trouble than what it's worth. Can we end up with the same person being assigned more than once? Yes, unfortunately, that's the nature of random. 

There's another piece to this is and that's the call to ensure that the owner from the AzureSdkOwners can actually be assigned to an issue within the repository. Even if there were two items in the list, if one of them can't be assigned an issue for that repository and the other can, the other would end up getting all the assignments anyways.